packages/flutter/lib/src/material/dialog.dart

```diff
@@ -950,6 +950,85 @@ Widget _buildMaterialDialogTransitions(BuildContext context, Animation<double> a
 /// Returns a [Future] that resolves to the value (if any) that was passed to
 /// [Navigator.pop] when the dialog was closed.
 ///
+/// ### State Restoration in Dialogs
+///
+/// Using this method with not enable state restoration for the dialog. In order
+/// to enable state restoration for a dialog, use [Navigator.restorablePush]
+/// or [Navigator.restorablePushNamed] with [DialogRoute].
+///
+/// For more information about state restoration, see [RestorationManager].
+///
+/// {@tool sample --template=freeform}
+///
+/// This sample demonstrates how to create a restorable Material dialog. This is
+/// accomplished by enabling state restoration by specifying
+/// [MaterialApp.restorationScopeId] and using [Navigator.restorablePush] to
+/// push [DialogRoute] when the button is tapped.
+///
+/// To test this out:
```

Comment:
Add: "on Android".

Actually, I am not super sure if we should document that android thing here. Could we just link to androids documentation?

Reply:
I can't find any official Android documentation that discusses how to do this. What's the reason for not being sure about documenting the Android thing? What would our alternatives be?

What about:
```
/// To test this out on Android, go to https://developer.android.com/studio/debug/dev-options
/// for instructions on how to enable Developer Mode on an Android device/simulator
/// and enable "Don't keep activities".
```